src/ChatGptNet/ChatGptClient.cs

```diff
@@ -133,7 +142,14 @@ public async IAsyncEnumerable<ChatGptResponse> AskStreamAsync(Guid conversationI
         {
             Model = model ?? options.DefaultModel,
             Messages = messages.ToArray(),
-            Stream = true
+            Stream = true,
+            Temperature = chatGptParameters?.Temperature ?? options.Temperature,
+            TopP = chatGptParameters?.TopP ?? options.TopP,
+            N = chatGptParameters?.N ?? options.N,
+            MaxTokens = chatGptParameters?.MaxTokens ?? options.MaxTokens,
+            PresencePenalty = chatGptParameters?.PresencePenalty ?? options.PresencePenalty,
+            FrequencyPenalty = chatGptParameters?.FrequencyPenalty ?? options.FrequencyPenalty,
+            LogitBias = chatGptParameters?.LogitBias ?? options.LogitBias
```

Comment:
Should I create this new method inside the ChatGptClient class or I need to create a new class (static?)?

Reply:
A new private method like `CreateRequest(...)` will be fine.

Reply:
I've created two method since the list of messages it's referenced forward

```csharp
private List<ChatGptMessage> AddMessageToConversationHistory(Guid conversationId, string message)
{
	ArgumentNullException.ThrowIfNull(message);

	// Ensures that conversationId isn't empty.
	if (conversationId == Guid.Empty)
	{
		conversationId = Guid.NewGuid();
	}

	// Checks whether a list of messages for the given conversationId already exists.
	var conversationHistory = cache.Get<IList<ChatGptMessage>>(conversationId);
	List<ChatGptMessage> messages = conversationHistory is not null ? new(conversationHistory) : new();

	messages.Add(new()
	{
		Role = ChatGptRoles.User,
		Content = message
	});

	return messages;
}

private ChatGptRequest CreateRequest(List<ChatGptMessage> messages, bool stream, string? model = null, ChatGptParameters? parameters = null)
{
	return new ChatGptRequest
	{
		Model = model ?? options.DefaultModel,
		Messages = messages.ToArray(),
		Stream = stream,
		Temperature = parameters?.Temperature ?? options.DefaultParameters.Temperature,
		TopP = parameters?.TopP ?? options.DefaultParameters.TopP,
		N = parameters?.N ?? options.DefaultParameters.N,
		MaxTokens = parameters?.MaxTokens ?? options.DefaultParameters.MaxTokens,
		PresencePenalty = parameters?.PresencePenalty ?? options.DefaultParameters.PresencePenalty,
		FrequencyPenalty = parameters?.FrequencyPenalty ?? options.DefaultParameters.FrequencyPenalty
	};
}
```

So inside the AskAsync/AskStreamAsync we can call those two methods:
```csharp
List<ChatGptMessage> messages = AddMessageToConversationHistory(conversationId, message);

var request = CreateRequest(messages, false, model, parameters);
```

Reply:
In the following lines of code:

```
ArgumentNullException.ThrowIfNull(message);

// Ensures that conversationId isn't empty.
if (conversationId == Guid.Empty)
{
	conversationId = Guid.NewGuid();
}
```

The checks are related to input validation, so I think it's better to leave them in both original methods, because otherwise they becomes a side-effect of the method. Moreover, the name `AddMessageToConversationHistory` isn't correct, because it doesn't actually add nothing to the history, it simply gets the list of messages, so it's better to call it just `GetMessages`.

For what concerns the `CreateRequest` method, remember that is better to have the `ChatGptParameters` before the model one.